### DISCUSSION TITLE
Role of lineGap parameter

### DISCUSSION THREAD

**Discussion by andriytyurnikov** [seek-oss/capsize#179] [Q&A] [answered]
As I am developing CSS-only (PostCSS, strictly speaking) port of Capsize, I went with simple case first, and assumed `lineGap:0`.
Not only many fonts have `lineGap:0`, but also few that don't (like `Helvetica Neue`) don't seem to be affected if I use `lineGap:0`.
Is it really necessary, or is it a just a legacy of old implementation, which was not using `display: table;` trick?

Please tell me what am I missing.

**Comment by michaeltaranto** [comment]
Hey @andriytyurnikov 👋  curious to understand more about what you are looking to achieve, as Capsize is "css-only" (unless you are calling it within your runtime code?).

The `lineGap` parameter gives consumers a different mental model for defining `line-height`, describing the desired "gap" between capital letters and baselines. It  pairs really nicely with `capHeight` as you are no longer defining `font-size` and `line-height` explicitly, rather the height of your capital letters and the space "between lines".

This works really well in a systems space, where you can better understand the relationship between your space scale and your typography spacing.

**Comment by andriytyurnikov** [comment]
Oh, I see. Hmm. I failed to grasp it from the start, and sometimes used padding. 
Basically, I set lineGap = 0, which allows shortest branch of your formula. 
Line-height setting still works on multiline text, and for single line text I added padding sometimes.

Hmm. I'll need some time to digest this.
Thanks!

**Comment by michaeltaranto** [comment, marked as answer]
Just to be clear, we are talking about the `lineGap` parameter to the Capsize API right?

There is also the `line gap` font metric internally, which is populated for some fonts, which I believe is additional line spacing beyond line height.

**Reply by andriytyurnikov** [reply]
Something, something… ubiquitous language 😂

Yes - absence of connection between those two things with same name is what I struggled with.
 Thanks for clarifying.

I guess traditional typography had “leading” for that, but, sure, digital tools , including Tailwind, have butchered traditional terminology long time ago.
